### PR TITLE
K8SPSMDB-604: Add `startupDelaySeconds` option for mongos liveness probe

### PR DESCRIPTION
**Pull request by pooknull**
[K8SPSMDB-604](https://jira.percona.com/browse/K8SPSMDB-604) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

https://jira.percona.com/browse/K8SPSMDB-604

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| storage | passed |
| upgrade-consistency | passed |
| one-pod | passed |
| self-healing | passed |
| self-healing-chaos | passed |
| limits | passed |
| monitoring-2-0 | failed |
| demand-backup | passed |
| scaling | passed |
| security-context | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| scheduled-backup | passed |
| rs-shard-migration | passed |
| arbiter | passed |
| demand-backup-sharded | failed |
| service-per-pod | passed |
| liveness | passed |
| upgrade | passed |
| smart-update | passed |
| upgrade-sharded | failed |
| version-service | passed |
| pitr | failed |
| users | passed |
| data-sharded | passed |
| non-voting | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/822/commits/e68254b043d352f844f642fd4afc52bb0b2a8eac
image: `perconalab/percona-server-mongodb-operator:PR-822-e68254b0`

File: pkg/apis/psmdb/v1/psmdb_defaults.go [hunk 1]
```diff
@@ -185,6 +189,12 @@ func (cr *PerconaServerMongoDB) CheckNSetDefaults(platform version.Platform, log
 						"--sslCAFile", "/etc/mongodb-ssl/ca.crt",
 						"--sslPEMKeyFile", "/tmp/tls.pem")
 			}
+
+			if cr.CompareVersion("1.4.0") >= 0 && !cr.Spec.Sharding.Mongos.LivenessProbe.CommandHas(startupDelaySecondsFlag) {
```
Comment (delgod) [inline pkg/apis/psmdb/v1/psmdb_defaults.go:193]: why 1.4.0 ?

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| limits | passed |
| scaling | passed |
| security-context | passed |
| rs-shard-migration | passed |
| storage | passed |
| self-healing | passed |
| self-healing-chaos | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| one-pod | passed |
| monitoring-2-0 | failed |
| upgrade-consistency | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| demand-backup-sharded | failed |
| upgrade | passed |
| upgrade-sharded | failed |
| arbiter | passed |
| service-per-pod | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| data-sharded | passed |
| non-voting | passed |
| pitr | passed |
| pitr-sharded | failed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/822/commits/e68254b043d352f844f642fd4afc52bb0b2a8eac
image: `perconalab/percona-server-mongodb-operator:PR-822-e68254b0`

**Review by delgod — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| storage | passed |
| one-pod | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| self-healing-chaos | passed |
| limits | passed |
| demand-backup | passed |
| monitoring-2-0 | failed |
| scaling | passed |
| security-context | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| scheduled-backup | passed |
| rs-shard-migration | passed |
| arbiter | passed |
| demand-backup-sharded | failed |
| service-per-pod | passed |
| liveness | passed |
| upgrade | passed |
| smart-update | passed |
| upgrade-sharded | failed |
| version-service | passed |
| pitr | passed |
| users | passed |
| data-sharded | passed |
| non-voting | passed |
| pitr-sharded | failed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/822/commits/e62980b612ee995835eabade8c6638b8bfb3cbb8
image: `perconalab/percona-server-mongodb-operator:PR-822-e62980b6`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| self-healing | passed |
| self-healing-chaos | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| init-deploy | passed |
| one-pod | passed |
| limits | passed |
| monitoring-2-0 | failed |
| scaling | passed |
| security-context | passed |
| rs-shard-migration | passed |
| upgrade-consistency | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| demand-backup-sharded | failed |
| upgrade | passed |
| upgrade-sharded | failed |
| arbiter | passed |
| service-per-pod | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| data-sharded | passed |
| non-voting | passed |
| pitr | passed |
| pitr-sharded | failed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/822/commits/e62980b612ee995835eabade8c6638b8bfb3cbb8
image: `perconalab/percona-server-mongodb-operator:PR-822-e62980b6`

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| one-pod | passed |
| storage | passed |
| init-deploy | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| self-healing-chaos | passed |
| limits | passed |
| monitoring-2-0 | failed |
| demand-backup | passed |
| scaling | passed |
| security-context | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| scheduled-backup | passed |
| rs-shard-migration | passed |
| arbiter | passed |
| demand-backup-sharded | failed |
| service-per-pod | passed |
| liveness | passed |
| upgrade | passed |
| smart-update | passed |
| upgrade-sharded | failed |
| version-service | passed |
| pitr | passed |
| users | passed |
| data-sharded | passed |
| non-voting | passed |
| pitr-sharded | failed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/822/commits/fd952d6e16e679f12fe5f3448b1b94b51125c36f
image: `perconalab/percona-server-mongodb-operator:PR-822-fd952d6e`

**Review by defbin — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| init-deploy | passed |
| limits | passed |
| scaling | passed |
| security-context | passed |
| rs-shard-migration | passed |
| storage | passed |
| self-healing | passed |
| self-healing-chaos | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| one-pod | passed |
| monitoring-2-0 | failed |
| upgrade-consistency | passed |
| demand-backup | passed |
| scheduled-backup | passed |
| demand-backup-sharded | failed |
| upgrade | passed |
| upgrade-sharded | failed |
| arbiter | passed |
| service-per-pod | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| users | passed |
| data-sharded | passed |
| non-voting | passed |
| pitr | passed |
| pitr-sharded | failed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/822/commits/fd952d6e16e679f12fe5f3448b1b94b51125c36f
image: `perconalab/percona-server-mongodb-operator:PR-822-fd952d6e`

**Review by delgod — APPROVED** [review]

**Review by defbin — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| storage | passed |
| one-pod | passed |
| init-deploy | passed |
| upgrade-consistency | passed |
| self-healing | passed |
| self-healing-chaos | passed |
| limits | passed |
| demand-backup | passed |
| monitoring-2-0 | passed |
| scaling | passed |
| arbiter | passed |
| security-context | passed |
| operator-self-healing | passed |
| operator-self-healing-chaos | passed |
| scheduled-backup | passed |
| rs-shard-migration | passed |
| service-per-pod | passed |
| demand-backup-sharded | passed |
| liveness | passed |
| smart-update | passed |
| version-service | passed |
| upgrade | passed |
| upgrade-sharded | passed |
| users | passed |
| pitr | passed |
| data-sharded | passed |
| non-voting | passed |
| pitr-sharded | passed |

commit: https://github.com/percona/percona-server-mongodb-operator/pull/822/commits/170bea6f96cc12db53ca60df409dbc034e53c707
image: `perconalab/percona-server-mongodb-operator:PR-822-170bea6f`

**Pull request merged**